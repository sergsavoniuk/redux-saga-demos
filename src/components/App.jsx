import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';

import Routes from 'components/Routes';
import Layout from 'components/Layout';
import AuthForm from 'components/AuthForm';
import Loader from 'components/Loader';
import { login, getIsAuthenticated } from 'redux/auth';

function getUsername() {
  return localStorage.getItem('username');
}

function App({ isAuthenticated, login, push }) {
  useEffect(() => {
    const username = getUsername();
    if (!isAuthenticated) {
      push('/auth');
      if (username != null) {
        login(username);
      }
    }
  }, [isAuthenticated]);

  let Component;
  if (!isAuthenticated) {
    Component = getUsername() != null ? <Loader /> : <AuthForm />;
  } else {
    Component = <Routes />;
  }

  return <Layout>{Component}</Layout>;
}

function mapStateToProps(state) {
  return {
    isAuthenticated: getIsAuthenticated(state),
  };
}

export default connect(
  mapStateToProps,
  { login, push },
)(App);
