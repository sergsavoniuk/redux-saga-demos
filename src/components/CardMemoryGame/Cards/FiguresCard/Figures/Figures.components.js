import styled from 'styled-components';

export const Title = styled.h2`
  width: 100%;
  text-transform: uppercase;
`;

export const Won = styled.span`
  margin: 0 5px;
`;
export const Lost = styled.span``;

export const Label = styled.i`
  vertical-align: top;
  font-size: 0.6em;
  color: #ffffff66;
  text-transform: uppercase;
`;
export const Value = styled.b``;
