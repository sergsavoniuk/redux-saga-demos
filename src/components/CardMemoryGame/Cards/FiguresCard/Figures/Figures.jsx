import React from 'react';

import {
  Title,
  Won,
  Lost,
  Abandoned,
  Label,
  Value,
} from './Figures.components';

function Figures({ won, lost, abandoned }) {
  return (
    <>
      <Title>
        Figures:
        <Won>
          <Value>{won}</Value>
          <Label>won</Label>
        </Won>
        <Lost>
          <Value>{lost}</Value>
          <Label>lost</Label>
        </Lost>
        <Abandoned>
          <Value>{abandoned}</Value>
          <Label>abandoned</Label>
        </Abandoned>
      </Title>
    </>
  );
}

export default Figures;
