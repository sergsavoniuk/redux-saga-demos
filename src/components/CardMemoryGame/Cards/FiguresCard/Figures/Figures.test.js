import React from 'react';
import { shallow } from 'enzyme';

import Figures from './Figures';
import { Row } from './Figures.components';

function getValueComponent(wrapper, index) {
  return wrapper
    .find(Row)
    .at(index)
    .children()
    .at(0); // Value component
}

describe('Testing Figures component', () => {
  test('renders correctly without props - used default values', () => {
    const wrapper = shallow(<Figures />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Row)).toHaveLength(3);

    expect(getValueComponent(wrapper, 0).text()).toMatch('0');
    expect(getValueComponent(wrapper, 1).text()).toMatch('0');
    expect(getValueComponent(wrapper, 2).text()).toMatch('0');
  });

  test('renders correctly with passed props', () => {
    const props = {
      won: 10,
      lost: 4,
      abandoned: 2,
    };

    const wrapper = shallow(<Figures {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Row)).toHaveLength(3);

    expect(getValueComponent(wrapper, 0).text()).toMatch(`${props.won}`);
    expect(getValueComponent(wrapper, 1).text()).toMatch(`${props.lost}`);
    expect(getValueComponent(wrapper, 2).text()).toMatch(`${props.abandoned}`);
  });
});
