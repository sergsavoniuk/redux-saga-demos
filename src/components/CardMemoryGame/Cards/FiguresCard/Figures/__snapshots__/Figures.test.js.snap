// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Figures component renders correctly with passed props 1`] = `
<styled.div>
  Figures:
  <styled.span>
    <styled.b>
      10
    </styled.b>
    <styled.i>
      won
    </styled.i>
  </styled.span>
  <styled.span>
    <styled.b>
      4
    </styled.b>
    <styled.i>
      lost
    </styled.i>
  </styled.span>
  <styled.span>
    <styled.b>
      2
    </styled.b>
    <styled.i>
      abandoned
    </styled.i>
  </styled.span>
</styled.div>
`;

exports[`Testing Figures component renders correctly without props - used default values 1`] = `
<styled.div>
  Figures:
  <styled.span>
    <styled.b>
      0
    </styled.b>
    <styled.i>
      won
    </styled.i>
  </styled.span>
  <styled.span>
    <styled.b>
      0
    </styled.b>
    <styled.i>
      lost
    </styled.i>
  </styled.span>
  <styled.span>
    <styled.b>
      0
    </styled.b>
    <styled.i>
      abandoned
    </styled.i>
  </styled.span>
</styled.div>
`;
